src/internal/common/idname/idname.go

```diff
@@ -0,0 +1,89 @@
+package idname
+
+import (
+	"strings"
+
+	"golang.org/x/exp/maps"
+)
+
+type Provider interface {
```

Comment:
since some of the functions below return `&Is{}` this should state what the behavior of the nil/empty `Provider` is